Classes/MutVoteSys.uc

```diff
@@ -1192,6 +1185,80 @@ function VS_Map LoadMapList(class<GameInfo> Game, name ListName) {
 	return ML.DuplicateList();
 }
 
+function AddMapsToMapList(array<string> MapArray, VS_MapList MapList, VS_Map IgnoreList) {
+	local string MapName;
+	local int i;
+
+	for (i = 0; i < MapArray.Length; i++)
+		if (MapArray[i] != "") {
+			MapName = MapArray[i];
+			if (Right(MapName, 4) ~= ".unr")
+				MapName = Left(MapName, Len(MapName) - 4); // we dont care about extension
+			if (!IsMapInMapList(MapName, IgnoreList))
+				MapList.AppendMap(MapName);
+		}
+}
+
+function AddMapPrefixesToMapList(array<string> MapPrefixArray, VS_MapList MapList, VS_Map IgnoreList) {
+	local string FirstMap;
+	local string MapName;
+	local int i;
+
+	for (i = 0; i < MapPrefixArray.Length; i++) {
+		if (MapPrefixArray[i] == "")
+			continue;
+
+		FirstMap = GetMapName(MapPrefixArray[i], "", 0);
+		if (FirstMap == "")
+			continue; // no maps with this prefix
+		MapName = FirstMap;
+
+		do {
+			if (!IsMapInMapList(MapName, IgnoreList))
+				MapList.AppendMap(Left(MapName, Len(MapName) - 4)); // we dont care about extension
+
+			MapName = GetMapName(MapPrefixArray[i], MapName, 1);
+		} until(MapName == FirstMap);
+	}
+}
+
+function AddMapListsToMapList(array<name> MapListArray, VS_MapList MapList, class<GameInfo> Game, VS_Map IgnoreList) {
+	local VS_Map IncludeList;
+	local int i;
+
+	for (i = 0; i < MapListArray.Length; i++) {
+		if (MapListArray[i] == '')
+			continue;
+
+		IncludeList = LoadMapList(Game, MapListArray[i]);
```

Comment:
This looks dangerous, because if `MapListArray[i]` is an empty name or `'None'`, all maps for Game will be included. This is fine when iterating through `IncludeList`, but not so much when iterating through `IgnoreList`.

I think LoadMapList should return `none` immediately if Game is `none` and ListName is empty.

You also shouldnt have to pass Game through all of these then and can just pass `none` to `LoadMapList()` then.